src/client.cpp

```diff
@@ -269,260 +269,99 @@ void ADDON_Destroy()
   m_CurStatus = ADDON_STATUS_UNKNOWN;
 }
 
-ADDON_STATUS ADDON_SetSetting(const char *settingName, const void *settingValue)
+ADDON_STATUS SetStringSetting(const string settingName, const char* settingValue, string currentValue, ADDON_STATUS returnValueIfChanged)
 {
-  string str = settingName;
-  if (str == "host")
+  string strSettingValue = settingValue;
+
+  if (strSettingValue != currentValue)
   {
-    string tmp_sHostname;
-    XBMC->Log(LOG_INFO, "%s - Changed Setting 'host' from %s to %s", __FUNCTION__, g_strHostname.c_str(), (const char*) settingValue);
-    tmp_sHostname = g_strHostname;
-    g_strHostname = (const char*) settingValue;
-    if (tmp_sHostname != g_strHostname)
-      return ADDON_STATUS_NEED_RESTART;
+    XBMC->Log(LOG_INFO, "%s - Changed Setting '%s' from %s to %s", __FUNCTION__, settingName.c_str(), currentValue.c_str(), strSettingValue.c_str());
+    currentValue = strSettingValue;
```

Comment:
As `currentValue` is a by-value parameter this modifies only your local copy! The parameter type has to changed. See my other comment with the corrected signature.